src/cdk-experimental/menu/menu-item-trigger.spec.ts

```diff
@@ -38,20 +40,288 @@ describe('MenuItemTrigger', () => {
+
+      triggers[0].toggle();
+      detectChanges();
+
+      expect(triggerSpy).toHaveBeenCalledTimes(1);
+      expect(closedSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should position the overlay below the trigger by default for a horizontal Menu', () => {
+      triggers[0].toggle();
+      detectChanges();
+
+      expect(Math.floor(nativeTriggers[0].getBoundingClientRect().bottom))
+        .withContext('MenuBar is horizontal by default')
+        .toEqual(Math.floor(nativeMenus[0].getBoundingClientRect().top));
+    });
+
+    it('should fallback to positioning the overlay above the trigger for horizontal Menu', () => {
+      nativeTriggers[0].style.position = 'fixed';
+      nativeTriggers[0].style.bottom = '0';
+
+      triggers[0].toggle();
+      detectChanges();
+
+      expect(Math.floor(nativeTriggers[0].getBoundingClientRect().top))
+        .withContext('trigger button position set to the bottom of the viewport')
+        .toEqual(Math.floor(nativeMenus[0].getBoundingClientRect().bottom));
+    });
+
+    it('should position nested submenu overlay to right by default in ltr layout', () => {
+      triggers[0].toggle();
+      detectChanges();
+      triggers[1].toggle();
+      detectChanges();
+
+      expect(Math.floor(nativeTriggers[1].getBoundingClientRect().right)).toEqual(
+        Math.floor(nativeMenus[1].getBoundingClientRect().left)
+      );
+    });
+
+    it('should fallback to positioning nested submenu overlay to the left in ltr layout', () => {
+      nativeTriggers[0].style.position = 'fixed';
+      nativeTriggers[0].style.right = '0';
+      triggers[0].toggle();
+      detectChanges();
+      triggers[1].toggle();
+      detectChanges();
+
+      expect(Math.floor(nativeTriggers[1].getBoundingClientRect().left))
+        .withContext('trigger positioned on the right')
+        .toEqual(Math.floor(nativeMenus[1].getBoundingClientRect().right));
+    });
+
+    it('should position nested submenu overlay to the left by default in rtl layout', () => {
+      setDocumentDirection('rtl');
+
+      triggers[0].toggle();
+      detectChanges();
+      triggers[1].toggle();
+      detectChanges();
+
+      expect(Math.floor(nativeTriggers[1].getBoundingClientRect().left)).toEqual(
+        Math.floor(nativeMenus[1].getBoundingClientRect().right)
+      );
+    });
+
+    it('should fallback to positioning nested submenu overlay to the right in rtl layout', () => {
+      setDocumentDirection('rtl');
+
+      nativeTriggers[0].style.position = 'fixed';
+      nativeTriggers[0].style.left = '0';
+      triggers[0].toggle();
+      detectChanges();
+
+      triggers[1].toggle();
+      detectChanges();
+
+      expect(Math.floor(nativeTriggers[1].getBoundingClientRect().right))
+        .withContext('trigger positioned on the left')
+        .toEqual(Math.floor(nativeMenus[1].getBoundingClientRect().left));
+    });
+
+    it('should position nested submenu at trigger level by default', () => {
```

Comment:
Is there a fallback test for this?

Reply:
No. My thinking is if the left/right fallback works testing left/right height would start testing the implementation of the Overlay class.